### PR TITLE
DS4 Icon Retrofit

### LINKED ISSUE #224: Icon: retrofit DS6 markup changes into DS4
The major changes are:

* removal of icon font (we can maybe *try* and keep it around for backwards compatibility, but I'm not opposed to killing it off)
* convert all current DS4 font-icons to background SVG
* removal of `svg-` prefix for inline SVG icons

If we do remove the icon font, we must remember to remove the reference from the [combo bundle](https://github.com/eBay/skin/blob/master/combo.browser.json#L6).

### PR DESCRIPTION
**Pull request by seangates**
## Description
DS4 icons are updated with variables and mixins. Allows DS4 background images and DS4 foreground images using inline SVG.

## Context
Current DS4 icons use an icon font which has limitations for our new eBayUI components. This PR brings the DS4 components forward and in sync with the DS6 icons. It uses both foreground (inline) SVG as well as background SVG using Base64 encoded strings.

## References
Closes #224 

## Screenshots
image
image
image
image

### CODE REVIEW

**Comment by seangates** [comment]
**Note**: The icons are now alphabetized. I could go through and have them mimic the old order, but having them alpha was just easier to track through the process of conversion.

**Comment by ianmcburnie** [comment]
@seangates The icons look considerably chunkier/bigger when compared to the current ds4 site. Or maybe it's just my eyes. Are those screenshots zoomed in?

**Comment by seangates** [comment]
@ianmcburnie The old versions were set smaller because they were using 12px font size. Since they are no longer using icon font they just use the set size from the SVG size. I can reset them to a size that scales with a smaller width. I simply just left them with the width as they were designed.

**Comment by seangates** [comment]
@ianmcburnie Also, the old ones were hard to actually see because the stroke widths were so stinking thin.

**Comment by ianmcburnie** [comment]
Ah, that makes sense. Let's leave them as is then for now. I like the chunkier look and feel so let's see how it goes. You are right, some of them were way too thin before!

**Comment by seangates** [comment]
@ianmcburnie Merged in 6.0.0. We should be good to go. Let me know if you see anything.

**Comment by ianmcburnie** [comment]
@seangates Have you tried updating any of the modules to use icons yet? Like expand button, notice, menu, etc? I presume that is the next step, but it might be nice to do just one in the PR if you didn't already. What do you think?

**Comment by seangates** [comment]
@ianmcburnie I got as far as verifying the Radio and Checkbox (based on your work). Yes, the other components would be next.

But, I can certainly work on doing one other. I'll give it a go.

**Comment by seangates** [comment]
@ianmcburnie Added the Base64 SVG vars to the menu. Please check once more.

File: src/less/menu/ds4/menu-base.less [hunk 1]
```diff
@@ -145,7 +145,7 @@ svg.fake-menu__status {
 
 span.menu__icon, // .menu__icon deprecated in v3.2.0
 span.fake-menu__icon { // .fake-menu__icon deprecated in v3.2.0
-    background-image: url('data:image/svg+xml,%3Csvg%20xmlns%3D%27http%3A//www.w3.org/2000/svg%27%20width%3D%2732%27%20height%3D%2732%27%3E%3Cpath%20fill%3D%22%230654ba%22%20d%3D%22M15.94%2025.294l-15.473-17.229h30.953l-7.742%208.616z%22%3E%3C/path%3E%3C/svg%3E');
+    background-image: url('data:image/svg+xml;base64,@{ds4-icon-arrow-down-base64}');
```
Comment (ianmcburnie) [inline src/less/menu/ds4/menu-base.less]: @seangates The `menu__icon` was deprecated in v3.2.0 in favor of using the `expand-btn`, so these changes wont be reflected on the website. Although it is still used on the test page.

Comment (seangates) [reply src/less/menu/ds4/menu-base.less]: We should change that, then.

Comment (ianmcburnie) [reply src/less/menu/ds4/menu-base.less]: Change what? Not sure I follow.

**Comment by ianmcburnie** [comment]
@seangates Sorry, I made more conflicts for you by rebasing the 5.1.1 patch from master into 6.0.0.

**Comment by seangates** [comment]
😞 Okay.

**Comment by seangates** [comment]
@ianmcburnie Should be all fixed, but please verify. I'm nervous I may have missed something.

**Comment by ianmcburnie** [comment]
Hmm, the commit history looks a bit strange. I see some of my older commits intermingled in the history of yours. But I will step through it all and make sure nothing is amiss.

**Comment by seangates** [comment]
@ianmcburnie I re-did the rebase. We should be good now.

**Comment by ianmcburnie** [comment]
Was the yarn.lock update required? Usually we try and keep these updates isolated. Seems a small amount of changes though, so it should be fine.

File: src/less/radio/base/radio.less [hunk 2]
```diff
@@ -4,7 +4,7 @@
 .radio {
     display: inline-flex;
     font-size: @font-size-16;
-    vertical-align: middle;
+    vertical-align: text-bottom;
```
Comment (ianmcburnie) [inline src/less/radio/base/radio.less]: I'm curious about why it needed changing to `text-bottom`...?

**Comment by seangates** [comment]
I actually thought I got rid of the yarn update. Let me look.

File: src/less/radio/base/radio.less (hunk 2 above)
Comment (seangates) [reply src/less/radio/base/radio.less]: It wast lining up properly with `middle`. You can try yourself and let me know what you think.

File: docs/_includes/ds4/svg.html [hunk 3]
```diff
@@ -38,8 +38,8 @@
         <path fill="#5ba71b" d="M16.009 32c-8.842 0-16.009-7.168-16.009-16.009s7.168-16.009 16.009-16.009c8.842 0 16.009 7.168 16.009 16.009 0 0.007 0 0.013 0 0.020-0.011 8.833-7.174 15.99-16.009 15.99 0 0 0 0 0 0zM16.009 2.319c0 0 0 0 0 0-7.561 0-13.69 6.129-13.69 13.69s6.129 13.69 13.69 13.69c7.561 0 13.69-6.129 13.69-13.69s-6.129-13.69-13.69-13.69z"></path>
         <path fill="#5ba71b" d="M14.841 23.448c-0.001 0-0.003 0-0.004 0-0.247 0-0.473-0.091-0.646-0.242l-6.232-5.305c-0.213-0.182-0.347-0.451-0.347-0.751 0-0.545 0.442-0.987 0.987-0.987 0.245 0 0.469 0.089 0.641 0.237l5.434 4.637 8.181-10.741c0.183-0.238 0.468-0.39 0.788-0.39 0.548 0 0.992 0.444 0.992 0.992 0 0.228-0.077 0.438-0.206 0.605l-8.81 11.573c-0.137 0.199-0.39 0.358-0.682 0.389l-0.116 0z"></path>
     </symbol>
-    <symbol id="svg-icon-close" viewBox="0 0 32 32">
-        <path d="M31.427 2.846l-2.387-2.387-13.084 13.082-13.082-13.082-2.386 2.387 13.082 13.082-13.082 13.084 2.386 2.386 13.082-13.082 13.084 13.082 2.386-2.386-13.084-13.084z"></path>
+    <symbol id="icon-close" viewBox="0 0 24 24">
```
Comment (ianmcburnie) [inline docs/_includes/ds4/svg.html]: Only one change to the DS4 inline SVG? I was expecting more. Shouldn't we have all the changes from `src/svg/ds4/icons.svg` in here too?

Comment (ianmcburnie) [reply docs/_includes/ds4/svg.html]: I'm not seeing any of the new inline icons when I run the website locally because of this I think.

Comment (seangates) [reply docs/_includes/ds4/svg.html]: Ugh. This is frustrating. I am 1000% positive this file was updated with everything.

Comment (seangates) [reply docs/_includes/ds4/svg.html]: I will update this file with a new commit, and let you know.

**Comment by ianmcburnie** [comment]
Checkbox and radio background icons should have a `#0654ba` fill for checked state, and `#ccc` fill for unchecked.

**Comment by ianmcburnie** [comment]
i’m not seeing some of the inline icons (the textbox with icon for example). I think because the `svg-` prefix needs removing from the `xlink:href`. Can you double check them all.

**Comment by seangates** [comment]
Wow, `#ccc` is really light!

**Comment by ianmcburnie** [comment]
Good point. We'll prob need to go to #767676 based on new WCAG color contrast guidelines. Is that what you currently had?

**Comment by seangates** [comment]
Yes, `#767676` is what I went with because it seemed appropriate.

**Comment by seangates** [comment]
But, the checked state I can fix, for sure.

**Comment by ianmcburnie** [comment]
A couple more missing icons, due to `svg-` prefix I think:

screen shot 2018-07-13 at 9 28 45 pm

We probably need to bump up the width and height from 8px too.

**Comment by seangates** [comment]
@ianmcburnie More than a couple. I was waiting to do each component in order to keep this PR size as small as possible. But, I have made the change everywhere. Let me know if you see any issues.

**Comment by ianmcburnie** [comment]
Yeah, we couldn't really leave the docs page in a broken state though with missing icons. 

Looks better now.

There are some alignment spacing issues now, but this was to be expected. I say let's do those adjustments in follow up PRs per component.

**Comment by seangates** [comment]
Let me know if we're all good now that I've fixed the checkbox/radio checked state colors.

File: src/less/less/ds4/mixins.less [hunk 4]
```diff
@@ -107,433 +107,415 @@
...
+    width: @width;
 }
 
-// Font-Icon mixins (font-icon is deprecated)
-//---------------------------------------
-
-.icon-arrowdown-mixin() {
-    content: "\e601";
+// base styles for BEM svg.icon block
+.inline-icon-base(@width: 1rem; @height: 1rem) {
```
Comment (ianmcburnie) [inline src/less/less/ds4/mixins.less]: I'm pretty sure I moved these 3 mixins out of ds4 into common, and I got rid of the dimension mixin (seems superfluous)... so maybe got lost in the merge?

**Comment by seangates** [comment]
@ianmcburnie Updated per that last comment on the mixins. Give it a go once more, por favor!

File: dist/less/ds6/mixins.less [hunk 5]
```diff
@@ -106,7 +106,7 @@
 
 // base styles for BEM svg.icon block
 // deprecated for the common mixin foreground-icon-base()
-.inline-icon-base(@width: 1rem; @height: 1rem) {
+.foreground-icon-base(@width: 1rem; @height: 1rem) {
```
Comment (ianmcburnie) [inline dist/less/ds6/mixins.less]: foreground-icon-base is common to ds4 and ds6, so shouldn't be showing up here. Maybe just need to run a `yarn build` again to regenerate this dist file...?

Comment (seangates) [reply dist/less/ds6/mixins.less]: Oh crap. That was a miss. Fixing ...

Comment (ianmcburnie) [reply dist/less/ds6/mixins.less]: I'm pretty sure I left inline-icon-base in here just for backwards compatibility (i.e. apps using this mixin). But as it's a major version change we can remove it if we want.

**Comment by seangates** [comment]
And ... another fix.

**Comment by seangates** [comment]
Thanks! And thanks, Ian, for your patience. More to come!

**Pull request merged**